TestStack.Dossier/DataSources/Picking/RandomItemSource.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace TestStack.Dossier.DataSources.Picking
+{
+    /// <summary>
+    /// Implements the random item strategy
+    /// </summary>
+    public class RandomItemSource<T> : DataSource<T>
+    {
+        /// <inheritdoc />
+        public RandomItemSource(IList<T> list)
+        {
+            Data = list;
+            Generator.StartIndex = 0;
+            Generator.ListSize = Data.Count;
+        }
+
+        /// <inheritdoc />
+        protected override IList<T> InitializeDataSource()
+        {
+            // This method will never be called as the list is set in the constructor.
+            throw new NotImplementedException();
```

Comment:
Alternatively the constructor could store the list in a private field and this method could return the lost? Then you might not need the internal set on the list?